evm-ext/src/solidity_abi.rs

```diff
@@ -355,3 +464,243 @@ fn force_to_u8(mv: MoveValue) -> u8 {
         unreachable!("Only call force_to_u8 with MoveValue::U8")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        alloy::primitives::Address,
+        aptos_gas_schedule::{InitialGasSchedule, LATEST_GAS_FEATURE_VERSION},
+        arbitrary::Unstructured,
+        move_core_types::value::{MoveFieldLayout, MoveStruct},
+        rand::{Rng, RngCore, rngs::ThreadRng, seq::SliceRandom},
+        std::time::{Duration, Instant},
+    };
+
+    #[test]
+    fn test_gas_costs() {
```

Comment:
Neat and fuzzy ✨ But I guess no other tests that would benefit from this approach?

Reply:
Potentially other tests would benefit! It might be nice to review and check. The difference here is I think this test could not have been written adequately without this approach. Getting gas costs right is important because once they are set we can only lower them as raising them might break existing code and we do not want to do that.